main/src/ca/uwaterloo/flix/language/errors/EntryPointError.scala

```diff
@@ -55,15 +53,34 @@ object EntryPointError {
          |
          |""".stripMargin
     })
+
+    override def loc: SourceLocation = SourceLocation.Unknown
   }
 
   /**
-    * Error indicating one or more arguments to an entry point function.
+    * An error raised to indicate that an entry point function is type-polymorphic.
+    *
+    * @param loc the location of the function symbol.
+    */
+  case class IllegalEntryPointPolymorphism(loc: SourceLocation) extends EntryPointError {
```

Comment:
-> IllegalPolymorphicEntryPoint (or IllegalNonMonomorphicEntryPoint)

Reply:
I've made it IllegalEntryPointTypeVariables. I think its nice to keep the language simple where possible